ivadomed/prepare_dataset.py

```diff
@@ -0,0 +1,215 @@
+def label2maskmap_gt(data, shape, c_dx=0, c_dy=0, radius=10, normalize=False):
+    """
+    Generate a Mask map from the coordenates
+    :param shape: dimension of output
+    :param data : input image
+    :param radius: is the radius of the gaussian function
+    :param normalize : bool for normalization.
+    :return: a MxN normalized array
+    """
+
+    # Our 2-dimensional distribution will be over variables X and Y
+    (M, N) = (shape[2], shape[1])
+    if len(data) <= 2:
+        # Output coordinates are reduced during post processing which poses a problem
+        data = [0, data[0], data[1]]
+    maskMap = []
+
+    x, y = data[2], data[1]
+
+    # Correct the labels
+    x += c_dx
+    y += c_dy
+
+    X = np.linspace(0, M - 1, M)
+    Y = np.linspace(0, N - 1, N)
+    X, Y = np.meshgrid(X, Y)
+
+    # Pack X and Y into a single 3-dimensional array
+    pos = np.empty(X.shape + (2,))
+    pos[:, :, 0] = X
+    pos[:, :, 1] = Y
+
+    # Mean vector and covariance matrix
+    mu = np.array([x, y])
+    Sigma = np.array([[radius, 0], [0, radius]])
+
+    # The distribution on the variables X, Y packed into pos.
+    Z = multivariate_gaussian(pos, mu, Sigma)
+
+    # Normalization
+    if normalize:
+        Z *= (1 / np.max(Z))
+    else:
+        # 8bit image values (the loss go to inf+)
+        Z *= (1 / np.max(Z))
+        Z = np.asarray(Z * 255, dtype=np.uint8)
+
+    maskMap.append(Z)
+
+    if len(maskMap) == 1:
+        maskMap = maskMap[0]
+
+    return np.asarray(maskMap)
+
+
+def multivariate_gaussian(pos, mu, Sigma):
+    """
+    Return the multivariate Gaussian distribution on array.
+
+    pos is an array constructed by packing the meshed arrays of variables
+    x_1, x_2, x_3, ..., x_k into its _last_ dimension.
+
+    """
+
+    n = mu.shape[0]
+    Sigma_det = np.linalg.det(Sigma)
+    Sigma_inv = np.linalg.inv(Sigma)
+    N = np.sqrt((2 * np.pi) ** n * Sigma_det)
+    # This einsum call calculates (x-mu)T.Sigma-1.(x-mu) in a vectorized
+    # way across all the input variables.
+    fac = np.einsum('...k,kl,...l->...', pos - mu, Sigma_inv, pos - mu)
+
+    return np.exp(-fac / 2) / N
+
+
+def add_zero_padding(img_list, x_val=512, y_val=512):
+    """
+    Add zero padding to each image in an array so they all have matching dimension.
+    :param img_list: list of input image to pad
+    :param x_val: shape of output alongside x axis
+    :param y_val: shape of output alongside y axis
+    :return: list of images
+    """
+    if type(img_list) != list:
+        img_list = [img_list]
+    img_zero_padding_list = []
+    for i in range(len(img_list)):
+        img = img_list[i]
+        img_tmp = np.zeros((x_val, y_val, 1), dtype=np.float64)
+        img_tmp[0:img.shape[0], 0:img.shape[1], 0] = img
+        img_zero_padding_list.append(img_tmp)
+
+    return img_zero_padding_list
+
+
+def mask2label(path_label, aim='full'):
```

Comment:
- move to utils
- if not used anymore (based on [previous suggestion](https://github.com/neuropoly/ivado-medical-imaging/pull/271/files#r436086717)), can remove

Reply:
Removed after previous suggestion